src/order/filter/basic.lean

```diff
@@ -535,6 +535,90 @@ begin
     exact ⟨p a, hpa, (s.inf p), ⟨⟨p, hp, le_refl _⟩, ht⟩⟩ }
 end
 
+/-! ### Eventually -/
+
+/-- `f.eventually p` or `∀ᶠ x in f, p x` mean that `{x | p x} ∈ f`. E.g., `∀ᶠ x in at_top, p x`
+means that `p` holds true for sufficiently large `x`. -/
+protected def eventually (p : α → Prop) (f : filter α) : Prop := {x | p x} ∈ f
+
+notation `∀ᶠ` binders ` in ` f `, ` r:(scoped p, filter.eventually p f) := r
+
+protected lemma eventually.and {p q : α → Prop} {f : filter α} :
+  f.eventually p → f.eventually q → ∀ᶠ x in f, p x ∧ q x :=
+inter_mem_sets
+
+lemma eventually_true (f : filter α) : ∀ᶠ x in f, true := univ_mem_sets
+
+lemma eventually_of_forall {p : α → Prop} (f : filter α) (hp : ∀ x, p x) :
+  ∀ᶠ x in f, p x :=
+univ_mem_sets' hp
+
+lemma eventually_false_iff_eq_bot {f : filter α} :
+  (∀ᶠ x in f, false) ↔ f = ⊥ :=
+empty_in_sets_eq_bot
+
+lemma eventually.mp {p q : α → Prop} {f : filter α} (hp : ∀ᶠ x in f, p x)
+  (hq : ∀ᶠ x in f, p x → q x) :
+  ∀ᶠ x in f, q x :=
+mp_sets hp hq
+
+lemma eventually.mono {p q : α → Prop} {f : filter α} (hp : ∀ᶠ x in f, p x)
+  (hq : ∀ x, p x → q x) :
+  ∀ᶠ x in f, q x :=
+hp.mp (f.eventually_of_forall hq)
+
+/-! ### Frequently -/
+
+/-- `f.frequently p` or `∀ᶠ x in f, p x` mean that `{x | ¬p x} ∉ f`. E.g., `∃ᶠ x in at_top, p x`
```

Comment:
There is still a typo in the notation. It reads `\forall^f` instead of `\exists^f`

Reply:
```suggestion
/-- `f.frequently p` or `∃ᶠ x in f, p x` mean that `{x | ¬p x} ∉ f`. E.g., `∃ᶠ x in at_top, p x`
```